### PR TITLE
Add missing Tests

### PR DESCRIPTION
**Pull request by julionc**
Cover test suites
`Redirect`, `SmartCollection`, `Collect`, `Custom Collection` tests

### CODE REVIEW

File: test/fixtures/smart_collection.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "body_html": null,
+    "published_at": null,
+    "sort_order": "alpha-asc",
+    "template_suffix": null,
+    "published_scope": "global",
+    "disjunctive": false,
+    "rules": [
+    ]
+  }
+}
```
Comment (kevinhughes27) [inline test/fixtures/smart_collection.json]: can you add some new lines here please!

**Comment by kevinhughes27** [comment]
Thanks for the tests!

**Comment by kevinhughes27** [comment]
Ping me when you update those newlines and I'll merge this.

**Comment by julionc** [comment]
Thanks for the tip! @kevinhughes27 
The PR is updated!

File: test/fixtures/smart_collection.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "disjunctive": false,
+    "rules": [
+      {
+        "column": "title",
+        "relation": "starts_with",
+        "condition": "mac"
+      }
+    ]
+  }
+}
```
Comment (kevinhughes27) [inline test/fixtures/smart_collection.json]: this one is still missing a newline

File: test/fixtures/redirect.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "redirect": {
+    "id": 979034150,
+    "path": "\/ipod",
+    "target": "\/pages\/itunes"
+  }
+}
```
Comment (kevinhughes27) [inline test/fixtures/redirect.json]: here too

**Comment by julionc** [comment]
Oops! I added the missing newlines. :sweat:
Quick question: I have seen a lot of tests using unit and spec style. Which one should I use?

**Comment by kevinhughes27** [comment]
I believe most of the tests here follow the `test ...` pattern so stick with that

**Pull request merged**